[19.0][MIG] stock, stock_account (#5547)
OCA/OpenUpgrade: openupgrade_scripts/scripts/stock/19.0.1.1/upgrade_analysis_work.txt

# NOTHING TO DO

stock        / stock.move               / group_id (many2one)           : DEL relation: procurement.group

# DONE: used to fill reference_ids in post-migration

stock        / stock.move               / inventory_name (char)         : NEW
stock        / stock.move               / name (char)                   : DEL required
stock        / stock.move               / package_level_id (many2one)   : DEL relation: stock.package_level

# NOTHING TO DO

stock        / stock.move               / packaging_uom_id (many2one)   : NEW relation: uom.uom, isfunction: function, stored

# DONE: filled from product_packaging_id in post-migration

stock        / stock.move               / packaging_uom_qty (float)     : NEW isfunction: function, stored
stock        / stock.move               / procurement_values (json)     : NEW stored: False

# NOTHING TO DO

stock        / stock.move               / product_packaging_id (many2one): DEL relation: product.packaging

# DONE: used to fill packaging_uom_id in post-migration

stock        / stock.move               / reference_ids (many2many)     : NEW relation: stock.reference

> # DONE: used to fill reference_ids in post-migration

Reviewer: ```suggestion
# DONE: filled from group_id in post-migration
```

Author: done

stock        / stock.move               / scrapped (boolean)            : DEL
stock        / stock.move.line          / description_picking (text)    : not stored anymore
stock        / stock.move.line          / description_picking (text)    : now related

> stock        / stock.move.line          / is_entire_pack (boolean)      : NEW

Reviewer: have you checked this?

Author: I didn't see any harm in having this unset, as the compute function for this is quite expensive. Now I'm running [`stock.picking#_check_entire_pack`](https://github.com/OCA/OCB/blob/19.0/addons/stock/models/stock_picking.py#L1304) on open pickings in end-migration

# NOTHING TO DO

stock        / stock.move.line          / package_level_id (many2one)   : DEL relation: stock.package_level

# DONE: renamed to package_history_id in pre-migration

stock        / stock.move.line          / reference (char)              : not stored anymore
stock        / stock.move.line          / result_package_id (many2one)  : relation is now 'stock.package' ('stock.quant.package') [nothing to do]

# NOTHING TO DO

stock        / stock.package            / child_package_dest_ids (one2many): NEW relation: stock.package
stock        / stock.package            / child_package_ids (one2many)  : NEW relation: stock.package
stock        / stock.package            / complete_name (char)          : NEW isfunction: function, stored
stock        / stock.package            / package_dest_id (many2one)    : NEW relation: stock.package
stock        / stock.package            / parent_package_id (many2one)  : NEW relation: stock.package
stock        / stock.package            / parent_path (char)            : NEW
stock        / stock.package.history    / outermost_dest_id (many2one)  : NEW relation: stock.package

> stock        / stock.package.history    / package_name (char)           : NEW required

Reviewer: this should be filled, as it is required

Author: done

stock        / stock.package.history    / parent_dest_id (many2one)     : NEW relation: stock.package
stock        / stock.package.history    / parent_dest_name (char)       : NEW
stock        / stock.package.history    / parent_orig_id (many2one)     : NEW relation: stock.package
stock        / stock.package.history    / parent_orig_name (char)       : NEW

# NOTHING TO DO: new feature

stock        / stock.package.history    / picking_ids (many2many)       : NEW relation: stock.picking

# DONE: set from stock.package_level#picking_id

stock        / stock.package.type       / package_use (selection)       : NEW required, selection_keys: ['disposable', 'reusable'], hasdefault: default

> # DONE: set to reusable if all previous stock.quant.package had this use TODO

Reviewer: ```suggestion
# DONE: set to reusable if all previous stock.quant.package had this use
```

Author: done

# DONE: renamed in pre-migration

stock        / stock.picking            / package_level_ids_details (one2many): DEL relation: stock.package_level
stock        / stock.picking            / shipping_weight (float)       : is now stored
stock        / stock.picking.type       / set_package_type (boolean)    : NEW hasdefault: default
stock        / stock.quant              / package_id (many2one)         : relation is now 'stock.package' ('stock.quant.package') [nothing to do]
stock        / stock.quant              / storage_category_id (many2one): not stored anymore
stock        / stock.quant.package      / _order                        : _order is now 'name, id' ('name')

# NOTHING TO DO

stock        / stock.quant.package      / package_use (selection)       : DEL required, selection_keys: ['disposable', 'reusable']

# DONE: used to set stock.package.type#package_use

stock        / stock.reference          / move_ids (many2many)          : NEW relation: stock.move

# DONE: renamed from stock_move_ids in pre-migration

stock        / stock.route              / package_type_selectable (boolean): NEW

> stock        / stock.route              / packaging_ids (many2many)     : DEL relation: product.packaging

Reviewer: this is done

Author: done

DEL ir.model.access: stock.access_product_template_attribute_line_manager
DEL ir.model.access: stock.access_product_template_stock_manager
DEL ir.model.access: stock.access_stock_change_product_qty
DEL ir.model.access: stock.access_stock_location_user
DEL ir.model.access: stock.access_stock_lot_report
DEL ir.model.access: stock.access_stock_package_level_all
DEL ir.model.access: stock.access_stock_package_level_stock_manager
DEL ir.model.access: stock.access_stock_package_level_stock_user
DEL ir.model.access: stock.access_stock_quant_package_all
DEL ir.model.access: stock.access_stock_quant_package_stock_manager
DEL ir.model.access: stock.access_stock_quant_package_stock_user
DEL ir.model.access: stock.access_stock_track_confirmation
DEL ir.model.access: stock.access_stock_track_line
DEL ir.model.access: stock.access_uom_category_stock_manager
DEL ir.model.access: stock.access_uom_uom_stock_manager
NEW ir.model.constraint: stock.constraint_stock_location_parent_path_id_idx
NEW ir.model.constraint: stock.constraint_stock_move_line_free_reservation_index
NEW ir.model.constraint: stock.constraint_stock_move_product_location_index

> DEL ir.model.constraint: stock.constraint_stock_warehouse_orderpoint_qty_multiple_check

Reviewer: I would delete safely this in pre

Author: done

DEL ir.ui.view: stock.duplicated_sn_warning
DEL ir.ui.view: stock.label_product_packaging_view
DEL ir.ui.view: stock.package_level_form_edit_view
DEL ir.ui.view: stock.package_level_form_view
DEL ir.ui.view: stock.package_level_tree_view_picking
DEL ir.ui.view: stock.procurement_group_form_view
DEL ir.ui.view: stock.product_packaging_form_view
DEL ir.ui.view: stock.product_packaging_tree_view
DEL ir.ui.view: stock.product_product_view_form_easy_inherit_stock
DEL ir.ui.view: stock.quant_package_search_view
DEL ir.ui.view: stock.search_customer_lot_filter
DEL ir.ui.view: stock.stock_lot_customer_report_view_list
DEL ir.ui.view: stock.view_change_product_quantity
DEL ir.ui.view: stock.view_quant_package_form
DEL ir.ui.view: stock.view_quant_package_kanban
DEL ir.ui.view: stock.view_quant_package_tree
DEL ir.ui.view: stock.view_stock_track_confirmation
NEW res.groups.privilege: stock.res_groups_privilege_inventory

# NOTHING TO DO

DEL stock.location: stock.stock_location_locations (noupdate)
DEL stock.location: stock.stock_location_locations_partner (noupdate)
DEL stock.location: stock.stock_location_locations_virtual (noupdate)

> # DONE: deleted in pre-migration

Reviewer: why in pre instead of post?

Author: good question
